source/common/upstream/upstream_impl.cc

```diff
@@ -1179,6 +1179,12 @@ ClusterInfoImpl::ResourceManagers::load(const envoy::config::cluster::v3::Cluste
   absl::optional<double> budget_percent;
   absl::optional<uint32_t> min_retry_concurrency;
   if (it != thresholds.cend()) {
+    // Catch case where max_retries is set but will be ignored!
```

Comment:
In the production system, who catches this exception?

Reply:
It's caught in `onConfigUpdate`:
https://github.com/envoyproxy/envoy/blob/master/source/common/upstream/cds_api_impl.cc#L53

At runtime the rejected update looks like this:
```
[2020-08-14 10:07:30.074][676864][info][upstream] [source/common/upstream/cds_api_impl.cc:64] cds: add 1 cluster(s), remove 1 cluster(s)
[2020-08-14 10:07:30.080][676864][warning][config] [source/common/config/filesystem_subscription_impl.cc:43] Filesystem config update rejected: Error adding/updating cluster(s) my-service: Cannot use both max_retries and retry_budget in cluster
```